### PR TITLE
Allow specifying custom edit handlers for pages - fixes #210

### PR DESCRIPTION
**Pull request by gasman**
This PR allows page classes to specify an `edit_handler` property, to override the default Content / Promote / Settings tabbed interface.

It also updates the default layout so that empty tabs are omitted, as requested in #485 + #972.

### CODE REVIEW

**Comment by kaedroho** [comment]
Looks good to me!

**Comment by zerolab** [comment]
:+1: 

This needs some documentation, other than a line in the changelog/release notes.

**Comment by gasman** [comment]
Documentation added. (Really ought to be in a separate 'Customisation' / 'Advanced' section along with 'Extending the WYSIWYG Editor' and 'Edit Handler API', but that's for another day...)

File: docs/core_components/pages/editing_api.rst [hunk 1]
```diff
@@ -376,6 +376,40 @@ For more on ``django-modelcluster``, visit `the django-modelcluster github proje
...
+        # field definitions omitted
+
+    BlogPage.content_panels = [
+        FieldPanel('title', classname="full title"),
+        FieldPanel('date'),
+        FieldPanel('body', classname="full"),
+    ]
+    BlogPage.sidebar_content_panels = [
+        SnippetChooserPanel('advert', Advert),
+        InlinePanel(BlogPage, 'related_links', label="Related links"),
```
Comment (kaedroho) [inline docs/core_components/pages/editing_api.rst]: This line needs updating

Comment (gasman) [inline docs/core_components/pages/editing_api.rst]: Good catch... updated.

File: wagtail/wagtailadmin/tests/test_pages_views.py [hunk 2]
```diff
@@ -158,6 +158,27 @@ def test_add_subpage_nonexistantparent(self):
     def test_create_simplepage(self):
         response = self.client.get(reverse('wagtailadmin_pages_create', args=('tests', 'simplepage', self.root_page.id)))
         self.assertEqual(response.status_code, 200)
+        self.assertContains(response, '<a href="#content" class="active">Content</a>')
```
Comment (kaedroho) [inline wagtail/wagtailadmin/tests/test_pages_views.py]: This ``TestCase`` is getting a bit big... Maybe we should break it down a bit some day.

**Comment by kaedroho** [comment]
:+1:

**Pull request merged**